Jwt hw

Oleh8 opened the pull request.

Oleh8 committed:
Added Jwt

Oleh8 committed:
Merge branch 'security-hw' into jwt-hw

Oleh8 committed:
Merge branch 'master' of github.com:Oleh8/book-store-app into jwt-hw

Oleh8 committed:
Merge branch 'jwt-hw' of github.com:Oleh8/book-store-app into jwt-hw

Oleh8 committed:
jwt-hw

Oleh8 committed:
Merge branch 'jwt-hw' of github.com:Oleh8/book-store-app into jwt-hw

Oleh8 committed:
jwt-added

Elena-Bruyako commented on src/main/java/bstore/bookstore/security/JwtAuthenticationFilter.java:
```diff
@@ -0,0 +1,52 @@
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
+        if (bearerToken != null && bearerToken.startsWith("Bearer")) {
```
create constant for Bearer 
```suggestion
        if (bearerToken != null && bearerToken.startsWith("Bearer ")) {
```

Elena-Bruyako commented on src/main/java/bstore/bookstore/security/JwtAuthenticationFilter.java:
```diff
@@ -0,0 +1,52 @@
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
+        if (bearerToken != null && bearerToken.startsWith("Bearer")) {
+            return bearerToken.substring(7);
```
```suggestion
            return bearerToken.substring(constant.length());
```

Elena-Bruyako reviewed (changes requested):
see comments

Oleh8 committed:
jwt-added

Oleh8 committed:
jwt-added

# Conflicts:
#	src/main/java/bstore/bookstore/security/JwtAuthenticationFilter.java

liliia-ponomarenko commented on src/main/java/bstore/bookstore/security/AuthenticationService.java:
```diff
@@ -0,0 +1,27 @@
+        String token = jwtUtil.generateToken(authentication.getName());
+        return new UserLoginResponseDto(token);
+    }
+
```
```suggestion
```

liliia-ponomarenko commented on src/main/java/bstore/bookstore/security/JwtAuthenticationFilter.java:
```diff
@@ -0,0 +1,53 @@
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
+        if (bearerToken != null && bearerToken.startsWith("Bearer ")) {
```
```suggestion
        if (bearerToken != null && bearerToken.startsWith(BEARER_TOKEN_PREFIX)) {
```

liliia-ponomarenko reviewed (approved):
Good job!
